simple_app/app_simple.py

```diff
@@ -28,28 +28,70 @@ def search():
-        query = query.upper()  # convert contents of search bar to all upper case
-        results: str = markdown(results.to_html(index=False, escape=False, max_rows=10,
-                                                classes='table table-dark table-bordered table-striped'))
+    try:
+        results: Optional[pd.DataFrame] = db.search_object(query, fmt='pandas')  # get the results for that object
+    except IndexError:
+        results = pd.DataFrame()
+    refresults: Optional[dict] = db.search_string(refquery, fmt='pandas', verbose=False)  # search all the strings
+    try:
+        refsources: pd.DataFrame = refresults['Sources']
+    except KeyError:
+        stringed_results: Optional[str] = None
     else:
-        results = None
-    return render_template('search.html', form=form,
-                           results=results, query=query)  # if everything not okay, return existing page as is
+        filtered_results: Optional[pd.DataFrame] = results.merge(refsources, on='source', suffixes=(None, 'extra'))
+        filtered_results.drop(columns=list(filtered_results.filter(regex='extra')), inplace=True)
+        sourcelinks: list = []  # empty list
+        if len(filtered_results):
+            for src in filtered_results.source.values:  # over every source in table
+                urllnk = quote(src)  # convert object name to url safe
+                srclnk = f'<a href="/solo_result/{urllnk}" target="_blank">{src}</a>'  # construct hyperlink
+                sourcelinks.append(srclnk)  # add that to list
+            filtered_results['source'] = sourcelinks  # update dataframe with the linked ones
+            query = query.upper()  # convert contents of search bar to all upper case
+            stringed_results = markdown(filtered_results.to_html(index=False, escape=False, max_rows=10,
+                                                                 classes='table table-dark '
+                                                                         'table-bordered table-striped'))
+        else:
+            stringed_results = None
+    return render_template('search.html', form=form, refquery=refquery,
+                           results=stringed_results, query=query)  # if everything not okay, return existing page as is
+
+
+@app_simple.route('/loose_search', methods=['GET', 'POST'])
+def loose_search():
+    """
+    Wrapping the search string function to search through all tables and return them
+    """
+    form = LooseSearchForm()  # main searchbar
+    if (query := form.search.data) is None:  # content in main searchbar
+        query = ''
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    results: Dict[str, pd.DataFrame] = db.search_string(query, fmt='pandas', verbose=False)  # search
+    resultsout = {}
+    if len(results):
+        for tabname, df in results.items():
+            if tabname in REFERENCE_TABLES:  # don't want the reference tables
```

Comment:
I disagree; I think in this case we do want the reference tables

Reply:
sure